yt/yt/python/yson/skiff/converter_skiff_to_python.cpp

```diff
@@ -318,7 +320,7 @@ class TStructSkiffToPythonConverter
         if (!obj) {
             THROW_ERROR_EXCEPTION("Failed to create field %Qv of class %Qv",
                 Description_,
-                PyType_.as_string())
+                PyTypeStr_)
```

Comment:
There was a bug here: you can't call `PyType_.as_string()` while there is an exception on the python stack.
If you try so, it will lead to a crash:

```python
yt-python-yt-wrapper-tests-py3_only: /home/echunaev/git/ytsaurus_v2/contrib/tools/python3/src/Objects/object.c:484: PyObject *PyObject_Str(PyObject *): Assertion `!_PyErr_Occurred(tstate)' failed.
Fatal Python error: Aborted

Thread 0x00007fb15373f700 (most recent call first):
  File "yt/python/yt/wrapper/transaction.py", line 411 in run
  File "contrib/tools/python3/src/Lib/threading.py", line 1045 in _bootstrap_inner
  File "contrib/tools/python3/src/Lib/threading.py", line 1002 in _bootstrap

Current thread 0x00007fb1748a93c0 (most recent call first):
  File "yt/python/yt/wrapper/table_commands.py", line 670 in _iterate
  File "yt/python/yt/wrapper/common.py", line 186 in merge_blobs_by_size
  File "yt/python/yt/wrapper/heavy_commands.py", line 281 in _iterator
  File "yt/python/yt/wrapper/retries.py", line 155 in action
  File "yt/python/yt/wrapper/retries.py", line 89 in run
  File "yt/python/yt/wrapper/retries.py", line 163 in run
  File "yt/python/yt/wrapper/heavy_commands.py", line 308 in __next__
  File "yt/python/yt/wrapper/response_stream.py", line 113 in _fetch
  File "yt/python/yt/wrapper/response_stream.py", line 23 in __init__
  File "yt/python/yt/wrapper/response_stream.py", line 161 in __init__
  File "yt/python/yt/wrapper/response_stream.py", line 169 in __init__
  File "yt/python/yt/wrapper/heavy_commands.py", line 397 in make_read_request
  File "yt/python/yt/wrapper/table_commands.py", line 841 in read_table
  File "yt/python/yt/wrapper/table_commands.py", line 872 in read_table_structured
  File "yt/python/yt/wrapper/tests/py3_only/test_typed_api.py", line 1668 in test_yt_dataclass_post_init
  File "library/python/pytest/plugins/ya.py", line 548 in pytest_pyfunc_call
  File "contrib/python/pluggy/py3/pluggy/_callers.py", line 77 in _multicall
  File "contrib/python/pluggy/py3/pluggy/_manager.py", line 115 in _hookexec
  File "contrib/python/pluggy/py3/pluggy/_hooks.py", line 493 in __call__
  File "contrib/python/pytest/py3/_pytest/python.py", line 1805 in runtest
  File "contrib/python/pytest/py3/_pytest/runner.py", line 169 in pytest_runtest_call
  File "contrib/python/pluggy/py3/pluggy/_callers.py", line 77 in _multicall
  File "contrib/python/pluggy/py3/pluggy/_manager.py", line 115 in _hookexec
  File "contrib/python/pluggy/py3/pluggy/_hooks.py", line 493 in __call__
  File "contrib/python/pytest/py3/_pytest/runner.py", line 262 in <lambda>
  File "contrib/python/pytest/py3/_pytest/runner.py", line 341 in from_call
  File "contrib/python/pytest/py3/_pytest/runner.py", line 261 in call_runtest_hook
  File "contrib/python/flaky/py3/flaky/flaky_pytest_plugin.py", line 138 in call_and_report
  File "contrib/python/pytest/py3/_pytest/runner.py", line 133 in runtestprotocol
  File "contrib/python/pytest/py3/_pytest/runner.py", line 114 in pytest_runtest_protocol
  File "contrib/python/flaky/py3/flaky/flaky_pytest_plugin.py", line 94 in pytest_runtest_protocol
  File "contrib/python/pluggy/py3/pluggy/_callers.py", line 77 in _multicall
  File "contrib/python/pluggy/py3/pluggy/_manager.py", line 115 in _hookexec
  File "contrib/python/pluggy/py3/pluggy/_hooks.py", line 493 in __call__
  File "contrib/python/pytest/py3/_pytest/main.py", line 350 in pytest_runtestloop
  File "contrib/python/pluggy/py3/pluggy/_callers.py", line 77 in _multicall
  File "contrib/python/pluggy/py3/pluggy/_manager.py", line 115 in _hookexec
  File "contrib/python/pluggy/py3/pluggy/_hooks.py", line 493 in __call__
  File "contrib/python/pytest/py3/_pytest/main.py", line 325 in _main
  File "contrib/python/pytest/py3/_pytest/main.py", line 271 in wrap_session
  File "contrib/python/pytest/py3/_pytest/main.py", line 318 in pytest_cmdline_main
  File "contrib/python/pluggy/py3/pluggy/_callers.py", line 77 in _multicall
  File "contrib/python/pluggy/py3/pluggy/_manager.py", line 115 in _hookexec
  File "contrib/python/pluggy/py3/pluggy/_hooks.py", line 493 in __call__
  File "contrib/python/pytest/py3/_pytest/config/__init__.py", line 169 in main
  File "library/python/pytest/main.py", line 97 in main

Extension modules: run_import_hook, __res, sitecustomize, library.python.strings.strings, simplejson._speedups, library.python.prctl.prctl, yt_yson_bindings.yson_lib, _brotli, six.moves._thread, numpy.core._multiarray_umath, numpy.core._multiarray_tests, numpy.linalg._umath_linalg, numpy.fft._pocketfft_internal, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._mt19937, numpy.random.mtrand, numpy.random._philox, numpy.random._pcg64, numpy.random._sfc64, numpy.random._generator, pyarrow.lib, pyarrow._parquet, pyarrow._fs, driver_lib, driver_rpc_lib, yt_tracing (total: 28)
2023-11-29 00:05:35,004 WARNING library.python.cores: Core dump dir doesn't exist: /coredumps
2023-11-29 00:05:35,004 WARNING library.python.cores: Core dump dir doesn't exist: /var/tmp/cores
Number of suites skipped by name: 2, by filter *test_yt_dataclass_post_init*

yt/python/yt/wrapper/tests/py3_only <py3test> [size:large] [tags: ya:external, ya:fat, ya:force_sandbox, ya:full_logs, ya:noretries, ya:yt, ya:ytexec]
------ sole chunk ran 1 test (total:11.19s - test:11.12s)
Test process was killed by SIGIOT signal. See logs for more info
Test wrapper's stderr tail:
on.prctl.prctl, yt_yson_bindings.yson_lib, _brotli, six.moves._thread, numpy.core._multiarray_umath, numpy.core._multiarray_tests, numpy.linalg._umath_linalg, numpy.fft._pocketfft_internal, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._mt19937, numpy.random.mtrand, numpy.random._philox, numpy.random._pcg64, numpy.random._sfc64, numpy.random._generator, pyarrow.lib, pyarrow._parquet, pyarrow._fs, driver_lib, driver_rpc_lib, yt_tracing (total: 28)
Log: /home/echunaev/git/ytsaurus_v2/yt/python/yt/wrapper/tests/py3_only/test-results/py3test/run_test.log
Logsdir: /home/echunaev/git/ytsaurus_v2/yt/python/yt/wrapper/tests/py3_only/test-results/py3test/testing_out_stuff
Stderr: /home/echunaev/git/ytsaurus_v2/yt/python/yt/wrapper/tests/py3_only/test-results/py3test/testing_out_stuff/stderr
[crashed] test_typed_api.py::TestTypedApi::test_yt_dataclass_post_init[v4] (chegoryu) [default-linux-x86_64-debug] (0.00s) Test crashed
Log: /home/echunaev/git/ytsaurus_v2/yt/python/yt/wrapper/tests/py3_only/test-results/py3test/testing_out_stuff/test_typed_api.py.TestTypedApi.test_yt_dataclass_post_init.v4.(chegoryu).log
Logsdir: /home/echunaev/git/ytsaurus_v2/yt/python/yt/wrapper/tests/py3_only/test-results/py3test/testing_out_stuff
------ FAIL: 1 - CRASHED yt/python/yt/wrapper/tests/py3_only

Total 1 suite:
        1 - FAIL
Total 1 test:
        1 - CRASHED
Failed

```